File: Examples/CSharp/Working-Shapes/RetrieveShapeInfo.cs
using System.IO;
using Aspose.Diagram;
using System;

namespace Aspose.Diagram.Examples.CSharp.Shapes
{
    public class RetrieveShapeInfo
    {
        public static void Run()
        {
            // ExStart:RetrieveShapeInfo
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Load diagram
            Diagram vsdDiagram = new Diagram(dataDir + "RetrieveShapeInfo.vsd");

            foreach (Aspose.Diagram.Shape shape in vsdDiagram.Pages[0].Shapes)
            {
                // Display information about the shapes
                Console.WriteLine("\nShape ID : " + shape.ID);
                Console.WriteLine("Name : " + shape.Name);
                Console.WriteLine("Master Shape : " + shape.Master.Name);
            }
            // ExEnd:RetrieveShapeInfo
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/RetrieveShapeInfo.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports System

Namespace Shapes
    Public Class RetrieveShapeInfo
        Public Shared Sub Run()
            ' ExStart:RetrieveShapeInfo
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()

            ' Load diagram
            Dim vsdDiagram As New Diagram(dataDir & "RetrieveShapeInfo.vsd")

            For Each shape As Aspose.Diagram.Shape In vsdDiagram.Pages(0).Shapes
                ' Display information about the shapes
                Console.WriteLine(Constants.vbLf & "Shape ID : " & shape.ID)
                Console.WriteLine("Name : " & shape.Name)
                Console.WriteLine("Master Shape : " & shape.Master.Name)
            Next shape
            ' ExEnd:RetrieveShapeInfo
        End Sub
    End Class
End Namespace